agent/token_store_test.go

```diff
@@ -0,0 +1,97 @@
+package agent
+
+import (
+	"testing"
+)
+
+func TestTokenStore_AgentToken(t *testing.T) {
+	t.Parallel()
+	tokens := new(TokenStore)
+
+	if got, want := tokens.GetTokenForAgent(), ""; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	// Precedence is agent token, user token, "".
+
+	tokens.UpdateUserToken("user")
+	if got, want := tokens.GetTokenForAgent(), "user"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateAgentToken("agent")
+	if got, want := tokens.GetTokenForAgent(), "agent"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateUserToken("")
+	if got, want := tokens.GetTokenForAgent(), "agent"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateAgentToken("")
+	if got, want := tokens.GetTokenForAgent(), ""; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+}
+
+func TestTokenStore_UserToken(t *testing.T) {
+	t.Parallel()
+	tokens := new(TokenStore)
+
+	if got, want := tokens.GetTokenForUser(), ""; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateUserToken("hello")
+	if got, want := tokens.GetTokenForUser(), "hello"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateUserToken("world")
+	if got, want := tokens.GetTokenForUser(), "world"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateUserToken("")
+	if got, want := tokens.GetTokenForUser(), ""; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+}
+
+func TestTokenStore_AgentMasterToken(t *testing.T) {
```

Comment:
This test feels a bit verbose. Not sure if a helper func would make it shorter but less readable though. You could add a helper to bail on wrong tokens but I'm not sure it really helps.

```go
isNot := func(toks ...string) {
    for _, tok := range toks {
        if tokens.IsAgentMasterToken(tok) {
            t.Fatalf("token %q is agent master token although it shouldn't be", tok)
        }
    }
}

...
isNot("", "nope")
```